geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/GCStatsMonitor.java

```diff
@@ -14,73 +14,76 @@
  */
 package org.apache.geode.management.internal.beans.stats;
 
+import java.util.concurrent.atomic.AtomicLong;
+
 import org.apache.geode.StatisticDescriptor;
 import org.apache.geode.internal.statistics.StatisticId;
 import org.apache.geode.internal.statistics.StatisticNotFoundException;
 import org.apache.geode.internal.statistics.StatisticsNotification;
 
 public class GCStatsMonitor extends MBeanStatsMonitor {
+  private AtomicLong collections = new AtomicLong(0);
+  private AtomicLong collectionTime = new AtomicLong(0);
 
-  private volatile long collections = 0;
-
-  private volatile long collectionTime = 0;
+  long getCollections() {
+    return collections.get();
+  }
 
+  long getCollectionTime() {
+    return collectionTime.get();
+  }
 
   public GCStatsMonitor(String name) {
     super(name);
   }
 
+  void decreasePrevValues(DefaultHashMap statsMap) {
+    collections.set(collections.get() - statsMap.get(StatsKey.VM_GC_STATS_COLLECTIONS).intValue());
```

Comment:
Calling get() and set() like this actually gives you the same race conditions that the volatile had.
What you want is to do a retry if the value you read when you called get() has changed by the time you call set.
You can do this using compareAndSet but in the class it would be much easier to just use "getAndAdd".
So in this case when you want to decrement do this:
collections.getAndAdd(-statsMap.get(StatsKey.VM_GC_STATS_COLLECTIONS).intValue());

Reply:
Missed that one, thanks for the heads up @dschneider-pivotal!!